flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNITCase.java

```diff
@@ -129,11 +150,20 @@ private void deployPerjob(YarnConfigOptions.UserJarInclusion userJarInclusion) t
 				assertThat(jobResult, is(notNullValue()));
 				assertThat(jobResult.getSerializedThrowable().isPresent(), is(false));
 
+				checkStagingDirectory(preUploadedFlinkBinary == null, applicationId);
+
 				waitApplicationFinishedElseKillIt(applicationId, yarnAppTerminateTimeout, yarnClusterDescriptor);
 			}
 		}
 	}
 
+	private void checkStagingDirectory(boolean shouldExist, ApplicationId appId) throws IOException {
+		final FileSystem fs = FileSystem.get(YARN_CONFIGURATION);
+		final Path stagingDirectory = new Path(fs.getHomeDirectory(), ".flink/" + appId.toString());
+		// If pre-uploaded flink is set correctly, the lib directory will not be uploaded to staging directory.
+		assertEquals(shouldExist, fs.exists(new Path(stagingDirectory, flinkLibFolder.getName())));
```

Comment:
We can move the comment as `message` parameter of `assertEquals`.

Reply:
Nice suggestion. I will move the message to `assertEquals`.